sup3r/preprocessing/data_handling/base.py

```diff
@@ -656,51 +629,105 @@ def preflight(self):
         if not check:
             logger.warning(msg)
             warnings.warn(msg)
 
-        msg = (
-            f'sample_shape[2] ({self.sample_shape[2]}) cannot be larger '
-            'than the number of time steps in the raw data '
-            f'({len(self.raw_time_index)}).')
+        msg = (f'sample_shape[2] ({self.sample_shape[2]}) cannot be larger '
+               'than the number of time steps in the raw data '
+               f'({len(self.raw_time_index)}).')
         if len(self.raw_time_index) < self.sample_shape[2]:
             logger.warning(msg)
             warnings.warn(msg)
 
-        msg = (
-            f'The requested time slice {self.temporal_slice} conflicts '
-            f'with the number of time steps ({len(self.raw_time_index)}) '
-            'in the raw data')
+        msg = (f'The requested time slice {self.temporal_slice} conflicts '
+               f'with the number of time steps ({len(self.raw_time_index)}) '
+               'in the raw data')
         t_slice_is_subset = start is not None and stop is not None
-        good_subset = (
-            t_slice_is_subset
-            and (stop - start <= len(self.raw_time_index))
-            and stop <= len(self.raw_time_index)
-            and start <= len(self.raw_time_index))
+        good_subset = (t_slice_is_subset
+                       and (stop - start <= len(self.raw_time_index))
+                       and stop <= len(self.raw_time_index)
+                       and start <= len(self.raw_time_index))
         if t_slice_is_subset and not good_subset:
             logger.error(msg)
             raise RuntimeError(msg)
 
-        msg = (
-            f'Initializing DataHandler {self.input_file_info}. '
-            f'Getting temporal range {self.time_index[0]!s} to '
-            f'{self.time_index[-1]!s} (inclusive) '
-            f'based on temporal_slice {self.temporal_slice}')
+        msg = (f'Initializing DataHandler {self.input_file_info}. '
+               f'Getting temporal range {self.time_index[0]!s} to '
+               f'{self.time_index[-1]!s} (inclusive) '
+               f'based on temporal_slice {self.temporal_slice}')
         logger.info(msg)
 
-        logger.info(
-            f'Using max_workers={self.max_workers}, '
-            f'norm_workers={self.norm_workers}, '
-            f'extract_workers={self.extract_workers}, '
-            f'compute_workers={self.compute_workers}, '
-            f'load_workers={self.load_workers}, '
-            f'ti_workers={self.ti_workers}')
+        logger.info(f'Using max_workers={self.max_workers}, '
+                    f'norm_workers={self.norm_workers}, '
+                    f'extract_workers={self.extract_workers}, '
+                    f'compute_workers={self.compute_workers}, '
+                    f'load_workers={self.load_workers}, '
+                    f'ti_workers={self.ti_workers}')
+
+    @staticmethod
+    def get_closest_lat_lon(lat_lon, target):
```

Comment:
This feels like it should be an instance method using self.lat_lon but im assuming you had to use it elsewhere in a static method. Maybe have a `lat_lon_3D_nearest()` utility method and then make this an instance method?

Also, if you're only letting target be a single coordinate then it's typically much faster to do the diff of lat/lon vs. the 2D arrays and find np.where(dist == np.min(dist)) instead of building a KDTree. The KDTree is typically only faster when you're doing lots of queries with lots of targets.

Reply:
I kept this as static since its called by class methods but changed to use diff. good catch on that.